### PR TITLE
Automation test for "Assays are messed up after folder rename" issue

### PR DESCRIPTION
**Pull request by labkey-sweta**
#### Rationale
 Test coverage : Issue 46473: Assays are messed up after folder rename
 https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=46473

1. Create an Assay project.
2. Create an assay design in that project.
3. Rename the project from Folder Management --> Rename and uncheck the Alias current name (recommended).
4. Manage Assay design --> Edit assay design should navigate to original assay design created and renamed folder.

#### Related Pull Requests
https://github.com/LabKey/platform/pull/3794

#### Changes
New test page src/org/labkey/test/pages/admin/RenameFolderPage.java

### CODE REVIEW

File: src/org/labkey/test/pages/admin/RenameFolderPage.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.openqa.selenium.WebElement;
+
+public class RenameFolderPage extends LabKeyPage<RenameFolderPage.ElementCache>
+{
+    public RenameFolderPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+    public static ReorderFoldersPage beginAt(WebDriverWrapper driver, String containerPath)
```
Comment (labkey-danield) [inline src/org/labkey/test/pages/admin/RenameFolderPage.java]: Why is the beginArt for the the **Rename**FolderPage returning a **Reorder**FolderPage?

File: src/org/labkey/test/pages/admin/RenameFolderPage.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        driver.beginAt(WebTestHelper.buildURL("admin", containerPath, "renameFolder"));
+        return new ReorderFoldersPage(driver.getDriver());
```
Comment (labkey-danield) [inline src/org/labkey/test/pages/admin/RenameFolderPage.java]: Do you need a `waitForPageLoad` or something like to make sure the page is loaded before returning?

**Review by labkey-danield — CHANGES_REQUESTED** [review]
Need some clarity on the return type for the beginAt method.

File: src/org/labkey/test/pages/admin/RenameFolderPage.java (hunk 1 above)
Comment (labkey-sweta) [reply src/org/labkey/test/pages/admin/RenameFolderPage.java]: good catch

File: src/org/labkey/test/pages/admin/RenameFolderPage.java (hunk 2 above)
Comment (labkey-sweta) [reply src/org/labkey/test/pages/admin/RenameFolderPage.java]: page is loading quickly

**Review by labkey-danield — APPROVED** [review]

**Pull request merged**